src/legacy/core_plugins/data/public/search/tabify/types.ts

```diff
@@ -17,8 +17,16 @@
  * under the License.
  */
 
-import './_get_columns';
-import './_buckets';
-import './_response_writer';
-import './_integration';
-describe('Tabify Agg Response', function() {});
+import { RangeFilterParams } from '../../../../../../plugins/data/public';
+
+/** @internal **/
+export interface TimeRange extends RangeFilterParams {
+  name: string;
+}
+
+/** @public **/
+export interface TabbedResponseWriterOptions {
+  metricsAtAllLevels: boolean;
+  partialRows: boolean;
+  timeRange?: { [key: string]: any };
```

Comment:
The `timeRange` that's passed here _should_ be the return value of the `getTime()` function (see esaggs L196).

So I believe you could type this using this `EsRangeFilter` interface: https://github.com/elastic/kibana/blob/master/src/plugins/data/common/es_query/filters/range_filter.ts#L66-L68

```suggestion
  timeRange?: EsRangeFilter['range'];
```